Server Crash When Generating New Terrain

# Server Crash When Generating New Terrain
The server crashed when trying to generate new chunks. The crash occurred specifically when the game was trying to place mangrove trees in new terrain, probably caused by Mob Grinding Utils's Delightful dirt updates.

# When Does It Happen
When exploring to new areas that haven't been generated yet
Specifically when mangrove trees are being placed during world generation (beside delightful dirt?)
I am playing on FTB Skies 2 which has some structure that meets both conditions : Mongrove + Delightful. It might not be a mangrove issue, it just mentionned in the crash log as is. That issue has also been reported on FTB Skies 2 issue tracker

# Error message
```
Caused by: java.lang.ClassCastException: class net.minecraft.server.level.WorldGenRegion cannot be cast to class net.minecraft.world.level.Level (net.minecraft.server.level.WorldGenRegion and net.minecraft.world.level.Level are in module minecraft@1.21.1 of loader 'TRANSFORMER' @e36bc01)
	at TRANSFORMER/mob_grinding_utils@1.1.9+mc1.21.1/mob_grinding_utils.blocks.BlockDelightfulDirt.updateShape(BlockDelightfulDirt.java:57) ~[mob_grinding_utils-1.1.9+mc1.21.1.jar%23747!/:1.1.9+mc1.21.1]
	
```

## Steps to reproduce
1 - Start a new world or travel to unexplored areas
2 - Generate a terrain that have 
2 - The server crashes when generating new terrain (island with Mangrove tree

Players discovering those chunks are unable to relog because they crash the server as soon as they log in.

Full crash log : https://gist.github.com/SebGrd/ab42c772048dcca69f18457865939608
Chunk IO Error report: https://gist.github.com/SebGrd/bcc78818e2122148625c018f79c9973a

## Mod Information
- Mod: Mob Grinding Utils
- Version: 1.1.9+mc1.21.1
- Minecraft: 1.21.1
- Loader: NeoForge 21.1.186
- Modpack: FTB Skies 2

## Additional Notes
This seems to be related to how the mod handles world generation
The crash happens in the BlockDelightfulDirt component

--- a/MobGrindingUtils/MobGrindingUtils/src/main/java/mob_grinding_utils/blocks/BlockDelightfulDirt.java
+++ b/MobGrindingUtils/MobGrindingUtils/src/main/java/mob_grinding_utils/blocks/BlockDelightfulDirt.java
@@ -36,12 +36,12 @@ public class BlockDelightfulDirt extends BlockDirtSpawner {
 		super(properties);
 	}
 
-	public boolean shouldSnowCap(Level level, BlockPos pos) {
+	public boolean shouldSnowCap(LevelAccessor level, BlockPos pos) {
 		// standard night ticks
-		return level.canSeeSkyFromBelowWater(pos) && (level.getDayTime() >= 13000 && level.getDayTime() <= 23000);
+		return level.canSeeSkyFromBelowWater(pos) && (level.dayTime() >= 13000 && level.dayTime() <= 23000);
 	}
 
-	public boolean shouldSpawnMob(Level level, BlockPos pos) {
+	public boolean shouldSpawnMob(LevelAccessor level, BlockPos pos) {
 		return level.getMaxLocalRawBrightness(pos.above()) >= 10 && level.getBlockState(pos.above()).isAir();
 	}
 
@@ -54,14 +54,14 @@ public class BlockDelightfulDirt extends BlockDirtSpawner {
 	@SuppressWarnings("deprecation")
 	@Override
 	public BlockState updateShape(BlockState stateIn, Direction facing, BlockState facingState, LevelAccessor level, BlockPos pos, BlockPos facingPos) {
-		if (shouldSnowCap((Level) level, pos) || shouldSpawnMob((Level) level, pos))
+		if (shouldSnowCap(level, pos) || shouldSpawnMob(level, pos))
 			level.scheduleTick(pos, this, Mth.nextInt(level.getRandom(), 20, 60));
 		return super.updateShape(stateIn, facing, facingState, level, pos, facingPos);
 	}
 
 	@Override
 	public void neighborChanged(BlockState state, Level level, BlockPos pos, Block blockIn, BlockPos fromPos, boolean isMoving) {
-		if (shouldSnowCap((Level) level, pos) || shouldSpawnMob((Level) level, pos))
+		if (shouldSnowCap(level, pos) || shouldSpawnMob(level, pos))
 			level.scheduleTick(pos, this, Mth.nextInt(level.random, 20, 60));
 	}
 

--- a/MobGrindingUtils/MobGrindingUtils/src/main/java/mob_grinding_utils/blocks/BlockDreadfulDirt.java
+++ b/MobGrindingUtils/MobGrindingUtils/src/main/java/mob_grinding_utils/blocks/BlockDreadfulDirt.java
@@ -36,12 +36,12 @@ public class BlockDreadfulDirt extends BlockDirtSpawner {
 		super(properties);
 	}
 
-	public boolean shouldCatchFire(Level level, BlockPos pos) {
+	public boolean shouldCatchFire(LevelAccessor level, BlockPos pos) {
 		// standard night to day ticks
-		return level.canSeeSkyFromBelowWater(pos) && (level.getDayTime() < 13000 || level.getDayTime() > 23000);
+		return level.canSeeSkyFromBelowWater(pos) && (level.dayTime() < 13000 || level.dayTime() > 23000);
 	}
 
-	public boolean shouldSpawnMob(Level level, BlockPos pos) {
+	public boolean shouldSpawnMob(LevelAccessor level, BlockPos pos) {
 		return level.getMaxLocalRawBrightness(pos.above()) < 5;
 	}
 
@@ -56,14 +56,14 @@ public class BlockDreadfulDirt extends BlockDirtSpawner {
 	@SuppressWarnings("deprecation")
 	@Override
 	public BlockState updateShape(BlockState stateIn, Direction facing, BlockState facingState, LevelAccessor level, BlockPos pos, BlockPos facingPos) {
-		if (shouldCatchFire((Level) level, pos) || shouldSpawnMob((Level) level, pos))
+		if (shouldCatchFire(level, pos) || shouldSpawnMob(level, pos))
 			level.scheduleTick(pos, this, Mth.nextInt(level.getRandom(), 20, 60));
 		return super.updateShape(stateIn, facing, facingState, level, pos, facingPos);
 	}
 
 	@Override
 	public void neighborChanged(BlockState state, Level level, BlockPos pos, Block blockIn, BlockPos fromPos, boolean isMoving) {
-		if (shouldCatchFire((Level) level, pos) || shouldSpawnMob((Level) level, pos))
+		if (shouldCatchFire(level, pos) || shouldSpawnMob(level, pos))
 			level.scheduleTick(pos, this, Mth.nextInt(level.getRandom(), 20, 60));
 	}
 

